airflow/providers/google/suite/transfers/sql_to_sheets.py

```diff
@@ -0,0 +1,140 @@
+    :param conn_id: the connection ID used to connect to the database.
+    :type sql_conn_id: str
+    :param parameters: The parameters to render the SQL query with.
+    :type parameters: dict or iterable
+    :param database: name of database which overwrite the defined one in connection
+    :type database: str
+    :param spreadsheet_range: The A1 notation of the values to retrieve.
+    :type spreadsheet_range: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = [
+        "sql",
+        "spreadsheet_id",
+        "spreadsheet_range",
+        "impersonation_chain",
+    ]
+
+    template_ext = (".sql",)
+    ui_color = "#a0e08c"
+
+    def __init__(
+        self,
+        *,
+        sql: str,
+        spreadsheet_id: str,
+        sql_conn_id: str,
+        parameters: Optional[Union[Mapping, Iterable]] = None,
+        database: str = None,
+        spreadsheet_range: str = "Sheet1",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.sql = sql
+        self.conn_id = sql_conn_id
+        self.database = database
+        self.parameters = parameters
+        self.gcp_conn_id = gcp_conn_id
+        self.spreadsheet_id = spreadsheet_id
+        self.spreadsheet_range = spreadsheet_range
+        self.delegate_to = delegate_to
+        self.impersonation_chain = impersonation_chain
+
+    def _data_prep(self, data):
+        for row in data:
+            item_list = []
+            for item in row:
+                if type(item) is datetime.date:
+                    item = item.strftime("%Y-%m-%d")
+                elif type(item) is datetime.datetime:
+                    item = item.strftime("%Y-%m-%d")
+                elif type(item) is decimal.Decimal:
+                    item = float(item)
```

Comment:
This `type(item) is <type>` kind of check is really not a good idea. Also you dropped the time part in a datetime, is it intended?

Something like this is likely better:

```suggestion
                if isinstance(item, (datetime.date, datetime.datetime)):
                    item = item.isoformat()
                elif isinstance(item, int):  # To exclude int from the number check.
                    pass
                elif isinstance(item, number.Number):
                    item = float(item)
```

Reply:
The dropped time was a mistake, fixed it now with your suggestion :)